analyses/cnv-frequencies/01-cnv-frequencies.py

```diff
@@ -0,0 +1,249 @@
+	cancer_group_df = all_tumors_df.groupby(["cancer_group"]).apply(func).reset_index()
+
+	# concat cancer_group_cohort_df and cancer_group_df and rremove duplicates
+	cancer_group_cohort_df = pd.concat([cancer_group_df, cancer_group_cohort_df], sort=False, ignore_index=True)
+	cancer_group_cohort_df.drop_duplicates(inplace=True, ignore_index=True)
+	# these subset dataframe is for testing and need to comment out 
+	#cancer_group_cohort_df = cancer_group_cohort_df[cancer_group_cohort_df.cancer_group.isin(["CNS Embryonal tumor", "Atypical Teratoid Rhabdoid Tumor"])]
+	return(cancer_group_cohort_df)
+
+
+def compute_variant_frequencies(all_tumors_df, primary_tumors_file, relapase_tumors_file, cancer_group_cohort_df):
+	# get independent primary tumor samples and subset from all tumor sample dataframe
+	tumor_dfs = {"all_tumors": all_tumors_df}
+	primary_tumors_samples_list = list(pd.read_csv(primary_tumors_file, sep="\t")
+	                              ["Kids_First_Biospecimen_ID"].unique())
+	primary_tumors_df = all_tumors_df[all_tumors_df.Kids_First_Biospecimen_ID.isin(primary_tumors_samples_list)].reset_index()
+	tumor_dfs["primary_tumors"] = primary_tumors_df
+
+	# get independent relapse tumor sample and subset from all tumor samples dataframe
+	relapse_tumors_samples_list = list(pd.read_csv(relapase_tumors_file, sep="\t")
+	                              ["Kids_First_Biospecimen_ID"].unique())
+	relapse_tumors_df = all_tumors_df[all_tumors_df.Kids_First_Biospecimen_ID.isin(relapse_tumors_samples_list)].reset_index()
+	tumor_dfs["relapse_tumors"] = relapse_tumors_df
+	          
+	# compute variant frequencies for each cancer group per cohort and  cancer group in cohorts
+	# for the overal dataset (all tumor samples)  and independent primary/replase tumor samples 
+	def func(x):
+	    d = {}
+	    d["Gene_Symbol"] = ",".join(x["gene_symbol"].unique())
+	    d["total_alterations"] = x["Kids_First_Participant_ID"].nunique()
+	    return pd.Series(d, index=["Gene_Symbol", "total_alterations"])
+	all_tumors_frequecy_dfs = []
+	primary_tumors_frequecy_dfs = []
+	relapse_tumors_frequecy_dfs = []
+	for index, row in cancer_group_cohort_df.iterrows():
+	    if row["num_samples"] > 5:
+	        for df_name, tumor_df in tumor_dfs.items():
+	            df = tumor_df[(tumor_df["cancer_group"] == row["cancer_group"]) & (tumor_df["cohort"] == row["cohort"])]
+	            if df.empty:
+	                continue
+	            num_pateints = df["Kids_First_Participant_ID"].nunique()
+	            df = df.groupby(["ensembl", "status"]).apply(func)
+	            df = df.rename_axis(["Gene_Ensembl_ID", "Variant_Type"]).reset_index()
+	            df["num_pateints"] = num_pateints
+	            for i, j in df.iterrows():
+	                df.at[i, "Total_Alterations/Patients_in_Dataset"] = f'{j["total_alterations"]}/{num_pateints}'
+	                df.at[i, "Frequency_in_Overall_Dataset"] = f'{(j["total_alterations"]/num_pateints)*100:.2f}%'
```

Comment:
Similarly, `f''` is not supported by the docker image.

```suggestion
	                df.at[i, "Total_Alterations/Patients_in_Dataset"] = '{}/{}'.format(j["total_alterations"], num_pateints)
	                df.at[i, "Frequency_in_Overall_Dataset"] = '{:.2f}%'.format((j["total_alterations"]/num_pateints)*100)
```